ORStools/utils/gui.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+"""GUI Utilities
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+
+__author__ = '(C) 2018 by Nyall Dawson'
+__date__ = '20/04/2018'
+__copyright__ = 'Copyright 2018, North Road'
+# This will get replaced with a git SHA1 when you do a git archive
+__revision__ = '$Format:%H$'
+
+import math
+import os
+import re
+from typing import (
+    Optional,
+    Union
+)
+
+from qgis.PyQt.QtCore import Qt
+from qgis.PyQt.QtGui import (
+    QIcon,
+    QFont,
+    QFontMetrics,
+    QImage,
+    QPixmap,
+    QFontDatabase,
+    QColor,
+    QPainter
+)
+from qgis.PyQt.QtSvg import QSvgRenderer
+from qgis.PyQt.QtWidgets import (
+    QMenu
+)
+from qgis.core import (
+    Qgis
+)
+from qgis.utils import iface
+
+from ORStools import RESOURCE_PREFIX
+from ORStools.utils import logger
+
+FONT_FAMILIES = ""
+
+FELT_STYLESHEET = """
+QDialog {
+    background-color: #ececec;
+    color: black;
+}
+QLabel {
+    color: black !important;
+    }
+
+QPushButton {
+    background: solid #3d521e !important;
+    color: white !important;
+}
+
+QLineEdit {
+    background: solid white;
+    color: black;
+}
+
+QProgressBar {
+    background: solid white;
+}
+
+"""
+
+
+class GuiUtils:
+    """
+    Utilities for GUI plugin components
+    """
+
+    APPLICATION_FONT_MAP = {}
+
+    @staticmethod
+    def set_link_color(html: str,
+                       wrap_color=True,
+                       color: Optional[Union[QColor, str]] = None) -> str:
+        """
+        Adds style tags to links in a HTML string for the standard link color
+        """
+        if color:
+            if isinstance(color, str):
+                color_string = color
+            else:
+                color_string = color.name()
+        else:
+            color_string = 'rgba(0,0,0,.3)'
+        res = re.sub(r'(<a href.*?)>',
+                     r'\1 style="color: {};">'.format(color_string),
+                     html)
+        if wrap_color:
+            res = '<span style="color: {};">{}</span>'.format(color_string,
+                                                              res)
+        return res
+
+    @staticmethod
+    def get_icon(icon: str) -> QIcon:
+        """
+        Returns a plugin icon
+        :param icon: icon name (svg file name)
+        :return: QIcon
+        """
+        path = GuiUtils.get_icon_svg(icon)
+        if not path:
+            return QIcon()
+
+        return QIcon(path)
+
+    @staticmethod
+    def get_icon_svg(icon: str) -> str:
```

Comment:
While this reads like it should do something with `svg`, the code does nothing - and rather has the functionality I'd expect in `get_icon`.